client/src/main/java/io/atomix/copycat/client/DefaultCopycatClient.java

```diff
@@ -150,6 +154,10 @@ private void onStateChange(Session.State state) {
       case UNSTABLE:
         setState(State.SUSPENDED);
         break;
+      case STALE:
+        setState(State.SUSPENDED);
+        this.close();
```

Comment:
not really, if user wanted recovery to continue happening why would they want to configure the unstabilityTimeout .
If they don't configure unstabilityTimeout then session would never reach STALE state and continue retrying to recover from UNSTABLE state.